pkg/webui/console/views/device/index.js

```diff
@@ -178,36 +180,43 @@ export default class Device extends React.Component {
     const asConfig = selectAsConfig()
     const hasAs =
       asConfig.enabled && application_server_address === getHostnameFromUrl(asConfig.base_url)
+    const hideMessages = !hasAs || !(mayScheduleDownlinks || maySendUplink)
+    const hidePayloadFormatters = !hasAs
+    const hideClaiming = !hasJs
 
     const basePath = `/applications/${appId}/devices/${devId}`
 
     // Prevent default redirect to uplink when tab is already open.
     const payloadFormattersLink = pathname.startsWith(`${basePath}/payload-formatters`)
       ? pathname
       : `${basePath}/payload-formatters`
+    const messagesLink = pathname.startsWith(`${basePath}/messages`)
+      ? pathname
+      : `${basePath}/messages`
 
     const tabs = [
       { title: sharedMessages.overview, name: 'overview', link: basePath },
       { title: sharedMessages.liveData, name: 'data', link: `${basePath}/data` },
       {
         title: sharedMessages.messages,
```

Comment:
```suggestion
        title: sharedMessages.messaging,
```

See my other comment.